http-clients-api/src/main/java/com/palantir/remoting/clients/ClientConfig.java

```diff
@@ -27,54 +27,76 @@
 import javax.net.ssl.X509TrustManager;
 import org.joda.time.Duration;
 
-// TODO(rfink) Is there anything JaxRs-specific in here? Can we reuse the same configuration for Retrofit2 clients?
+// TODO(rfink) Use immutables? How is this different from ServiceConfiguration? Is it just the SslConfiguration?
 public final class ClientConfig {
+
+    private static final Duration CONNECT_TIMEOUT = Duration.standardSeconds(10);
+    private static final Duration READ_TIMEOUT = Duration.standardMinutes(10);
+    private static final Duration WRITE_TIMEOUT = Duration.standardMinutes(10);
+    private static final int MAX_NUM_RETRIES = 1;
+
     private Optional<SSLSocketFactory> thisSslSocketFactory = Optional.absent();
     private Optional<X509TrustManager> thisTrustManager = Optional.absent();
     private Optional<Duration> thisConnectTimeout = Optional.absent();
     private Optional<Duration> thisReadTimeout = Optional.absent();
+    private Optional<Duration> thisWriteTimeout = Optional.absent();
```

Comment:
otherwise field names shadow method parameters. as per the TODO above, I think this class will go away and be replaced by an immutables class.